File: scripts/hooks/git-push-reminder.js
#!/usr/bin/env node
/**
 * git-push-reminder.js — Remind to review before git push
 *
 * Advisory (exit 0). Source: ECC git-push-reminder pattern.
 */

const fs = require('fs');

try {
  const raw = fs.readFileSync(0, 'utf8');
  const input = JSON.parse(raw);
  const command = (input.tool_input && input.tool_input.command) || '';

  if (/\bgit\s+push\b/.test(command)) {
    console.log(
      'SuperOPC: About to push. Checklist:\n' +
      '  1. Tests pass?\n' +
      '  2. No debug statements left?\n' +
      '  3. Commit messages follow Conventional Commits?\n' +
      '  4. No secrets in committed files?'
    );
  }
} catch (e) {
  // Silent pass
}

process.exit(0);

File: scripts/hooks/git_push_reminder.py
#!/usr/bin/env python3

from __future__ import annotations

import re
import sys

from common import get_tool_input, read_stdin_json, write_message


def main() -> int:
    payload = read_stdin_json()
    command = get_tool_input(payload).get("command", "")

    if isinstance(command, str) and re.search(r"\bgit\s+push\b", command):
        write_message(
            "SuperOPC: About to push. Checklist:\n"
            "  1. Tests pass?\n"
            "  2. No debug statements left?\n"
            "  3. Commit messages follow Conventional Commits?\n"
            "  4. No secrets in committed files?"
        )

    return 0


if __name__ == "__main__":
    sys.exit(main())
